core/modules/main/templates/_findrelatedissues.inc.php

```diff
@@ -1,29 +1,41 @@
+<?php if ($grouped_issues): ?>
+
+    <div class="header_div"><?= __('The following issues matched your search'); ?>:</div>
+    <span class="faded_out"><?= __('Either use the checkboxes and press the "%relate_these_issues"-button below or click any issues in the list, and select an action.', array('%relate_these_issues' => __('Relate these issues'))); ?></span>
+    <form id="viewissue_relate_issues_form" action="<?= make_url('viewissue_relate_issues', array('project_key' => $issue->getProject()->getKey(), 'issue_id' => $issue->getID())); ?>" method="post" accept-charset="<?= \thebuggenie\core\framework\Settings::getCharset(); ?>" onsubmit="TBG.Issues.relate('<?= make_url('viewissue_relate_issues', array('project_key' => $issue->getProject()->getKey(), 'issue_id' => $issue->getID())); ?>');return false;">
+
         <div style="height: 400px; overflow: auto">
-        <table style="width: auto; border: 0;" cellpadding="0" cellspacing="0">
-            <?php foreach ($issues as $aissue): ?>
-                <?php if ($aissue->getID() == $issue->getID()): continue; endif; ?>
-                <tr>
-                    <td style="width: 20px;"><input type="checkbox" value="<?php echo $aissue->getID(); ?>" name="relate_issues[<?php echo $aissue->getID(); ?>]" id="relate_issue_<?php echo $aissue->getID(); ?>"></td>
-                    <td class="issue_title">
-                        <label for="relate_issue_<?php echo $aissue->getID(); ?>" style="font-weight: normal;">[<?php if ($aissue->getState() == \thebuggenie\core\entities\Issue::STATE_OPEN): echo __('OPEN'); else: echo __('CLOSED'); endif; ?>] <?php echo $aissue->getFormattedTitle(); ?></label>
-                    </td>
-                </tr>
+            <?php foreach($grouped_issues as $project => $matched_issues): ?>
+                <?php if ($issue->getProject()->getName() == $project): ?>
+                    <div class="header_div smaller"><?= __('Current project') ?></div>
+                <?php else: ?>
+                    <div class="header_div smaller"><?= $project ?></div>
+                <?php endif; ?>
+
+                <table style="width: auto; border: 0;" cellpadding="0" cellspacing="0">
+
+                    <?php foreach($matched_issues as $matched_issue): ?>
+                        <tr>
+                            <td style="width: 20px;"><input type="checkbox" value="<?= $matched_issue->getID(); ?>" name="relate_issues[<?= $matched_issue->getID(); ?>]" id="relate_issue_<?= $matched_issue->getID(); ?>"></td>
+                            <td class="issue_title">
+                                <label for="relate_issue_<?= $matched_issue->getID(); ?>" style="font-weight: normal;">[<?= __($matched_issue->getStateAsText()); ?>] <?= $matched_issue->getFormattedTitle(); ?></label>
```

Comment:
I'd avoid dynamic translation casts as much as possible, mainly because they are hard to maintain. Doing 
`<?= ($issue->isOpen()) ? __('Open') : __('Closed'); ?>` also reads easier. Also, `OPEN` and `Open` wouldn't need two translations, uppercase can be achieved with a small bit of css instead, if required :)

Reply:
Ok, I'll try to do that one instead - I primarily wanted to do away with the ugly bunch of php if/else blocks, so your proposed solution looks quite nice. I'll look into the CSS part too.

Reply:
Ok, this one has been fixed. I'll wait for your reply on the other issue first before making that particular change :)